application/application.go

```diff
@@ -0,0 +1,101 @@
+package application
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/josephroberts/edge-node-manager/config"
+	"github.com/josephroberts/edge-node-manager/database"
+	"github.com/josephroberts/edge-node-manager/device"
+	"github.com/josephroberts/edge-node-manager/helper"
+	"github.com/josephroberts/edge-node-manager/proxyvisor"
+)
+
+type Application struct {
+	UUID       string
+	DeviceType *device.Type
+}
+
+func (a *Application) Process() error {
+	log.Printf("-----------------------------------------------------------------\r\n")
+	log.Printf("Application UUID: %s\r\n", a.UUID)
+	log.Printf("Application device type: %s\r\n", a.DeviceType.Device)
+	log.Printf("Application radio type: %s\r\n", a.DeviceType.Radio)
+
+	// Get latest firmware .zip and commit
+	application, commit, err := helper.GetApplication(config.GetPersistantDirectory(), a.UUID)
+	if err != nil {
+		log.Printf("No application location or commit found\r\n")
+	} else {
+		log.Printf("Application location: %s\r\n", application)
+		log.Printf("Application commit: %s\r\n", commit)
+	}
+
+	// Load applicaion devices i.e. already provisioned
+	applicationDevices, err := database.LoadDevices(a.UUID, a.DeviceType)
+	if err != nil {
+		return fmt.Errorf("Unable to load devices: %v", err)
+	}
+	log.Printf("Application devices found: %d\r\n", len(applicationDevices))
```

Comment:
The app is pretty noisy, it might be worth looking at 3rd-party logging packages and choosing one with log levels then putting stuff like this at a 'debug' level. Otherwise any logs generated from this are going to get messy, fast.